ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -270,4 +265,36 @@ instance Ord sv => ChainOrder (SimpleChainOrder sv) where
 
   preferCandidate _cfg ours cand = ours < cand
 
+deriving via SimpleChainOrder () instance ChainOrder ()
 deriving via SimpleChainOrder BlockNo instance ChainOrder BlockNo
+
+data SelectView p = SelectView
+  { svBlockNo :: !BlockNo
+  , svTiebreakerView :: !(TiebreakerView p)
+  }
+  deriving stock Generic
```

Comment:
Probably unrelated to this PR, but I was wondering if we could elaborate on why the block number (which will become a weigth) and the notion of a tie-breaker cannot be unified into a single weight notion.

Reply:
I am not exactly sure what unification you have in mind, is it something different than what we have in current main, where everybody defines SelectView as a "single" thing determining the chain order for themselves?

> Do we want to keep the block number and tiebreaker separately because of the separation that we currently have between sorting candidates and preferring a chain?

It is unrelated to that distinction; the motivation is 

 1. (minor) to make it explicit that `BlockNo` must be the primary way of comparing chains; even before this PR, `ConsensusProtocol`s didn't have the freedom to use arbitrary `SelectView`s, and
 2. nicely supports weighted chain comparisons, see "Primary motivation" in the PR description, as well as #1594.

Reply:
FTR, we discussed this offline with @amesgen. The key constraint we have is that we rely on a transitive `Ord` instance when selecting candidates. Therefore, we cannot unify the sorting of `Views` in a single function.  I think that we have is indeed two comparison functions that operate on `SelectView`s. It's just that we're explicit about the part of the field that is used for breaking ties (which I think makes sense)